var/spack/repos/builtin/packages/silo/package.py

```diff
@@ -67,6 +71,9 @@ def flag_handler(self, name, flags):
                 flags.append(self.compiler.cxx_pic_flag)
             elif name == 'fcflags':
                 flags.append(self.compiler.fc_pic_flag)
+        if name == 'cflags':
+            if spec.satisfies('%apple-clang'):
```

Comment:
This will undoubtedly happen with the other clang compilers too: can you change to
```suggestion
            if spec.compiler.name in ['clang', 'apple-clang']:
```
and then others can append to this list as needed? Once the implicit declaration is fixed we can restrict the HDF5 version as well.